luigi2.h

```diff
@@ -3041,6 +3200,48 @@ int _UICodeMessage(UIElement *element, UIMessage message, int di, void *dp) {
 	return 0;
 }
 
+#if UI_UNICODE
+void UICodeMoveCaret(UICode *code, bool backward, bool word) {
```

Comment:
Again, if you have separate defines for the string operations I think you won't need to duplicate this entire function's logic?